x-pack/plugins/maps/public/embeddable/map_embeddable.tsx

```diff
@@ -568,14 +572,21 @@ export class MapEmbeddable
     } as ActionExecutionContext;
   };
 
-  setMapExtentFilter() {
-    const state = this._savedMap.getStore().getState();
-    const mapExtent = getMapExtent(state);
-    const geoFieldNames = getGeoFieldNames(state);
-    const center = getMapCenter(state);
-    const zoom = getMapZoom(state);
+  // Timing bug for dashboard with multiple maps with synchronized movement and filter by map extent enabled
+  // When moving map with filterByMapExtent:false, previous map extent filter(s) does not get removed
+  // Cuased by syncDashboardContainerInput applyContainerChangesToState.
+  //   1) _setMapExtentFilter executes ACTION_GLOBAL_APPLY_FILTER action,
+  //      removing previous map extent filter and adding new map extent filter
+  //   2) applyContainerChangesToState then re-adds stale input.filters (which contains previous map extent filter)
+  // Add debounce to fix timing issue.
+  //   1) applyContainerChangesToState now runs first and does its thing
+  //   2) _setMapExtentFilter executes ACTION_GLOBAL_APPLY_FILTER action,
+  //      removing previous map extent filter and adding new map extent filter
```

Comment:
Is there anything we can do to suppress the UI updates between adding the new filter and removing the previous filter? It seems to cause a distracting flash.

Also, I'm experiencing some cases where the previous filter is not getting removed. Might be easier to show you in sync.

https://user-images.githubusercontent.com/1638483/178532069-080e88fd-18b0-4c36-8fad-66ad6cfb92e7.mp4